# Bazaar merge directive format 2 (Bazaar 0.90)
# revision_id: gbenison@gmail.com-20080728153014-9cmh3qp4879910dg
# target_branch: http://arch.gna.org/guile-gnome/bzr/gtk/
# testament_sha1: 1dd74795aba452934a0810f47bfe1ad34b958e4b
# timestamp: 2008-07-28 08:36:30 -0700
# base_revision_id: wingo@pobox.com-20080703151735-je9a72h0nlq3eere
# 
# Begin patch
=== modified file 'gnome/gw/gdk-support.c'
--- gtk/gnome/gw/gdk-support.c	2008-04-24 11:17:07 +0000
+++ gtk/gnome/gw/gdk-support.c	2008-07-28 15:30:14 +0000
@@ -228,11 +228,22 @@
     return ret;
 }
 
+/*
+ * Allocate a new GdkColor* which must be freed using
+ * gdk_color_free().
+ */
+static GdkColor*
+gdk_color_new()
+{
+  GdkColor tmp;
+  return gdk_color_copy(&tmp);
+}
+
 GdkColor*
 scm_scm_to_gdk_color (SCM scm)
 #define FUNC_NAME "%scm->gdk-rectangle"
 {
-    GdkColor *ret = g_new0 (GdkColor, 1);
+    GdkColor *ret = gdk_color_new();
     
     if (scm_is_string (scm)) {
         char *chars;

# Begin bundle
IyBCYXphYXIgcmV2aXNpb24gYnVuZGxlIHY0CiMKQlpoOTFBWSZTWZjPJ74AAbHfgEAQUXP//3qA
AAC////6UAPZV7KKmigbBkSKemTJNjQBMKepo9RkZNGQ00MAAAABoAAAAAEomgmgaCTNBR6NQHpD
0TTT1GjDAAAAAaAAAAABJIQ0j0TBJ5JiGmAppkZBoZHetMt0dHeL+Zs1essYZwmiipWtmW5noD6t
JP2bTdtsv1JJPzNyeehztVVSPROr6m0YKZQFaOWu+z06Rud5U/7D6/HzSd7loXjOuOtTyhUB1+CF
uquimbPJylUSYlaqGEDm3OnfJaLqj2yQ6r3QgKpKtA0pmW2HITLziKOTjm/fVLt6fxQHZtkGFTeB
FuHcCiRWzsuJDoOcpYPAVFiEB8InMDl4naQHikc4a1XMxmYpaVYnPm2xrL3exYl0MEuEnlJO+JM5
UFYwKS8BCwsgYhIWS2kuWGPMseMJO8ncXBIuJJ/X3ZRKMVCSIGaSULHRgmhOe9tGooKytZCskgmJ
VS3QycULqKoxuNQ40lJoxtNxRyHHy7M0nUOE7BicXaUrMD5NrgsTCJtL8C5FPpE4MOJhg9nMJwT3
DIsLzYUpCYlqPMWnEscKgQYSs2YjyQ6qxhXiYYlgxArte4UW+sCwwi2FpYQLks5oURZypVRHaVFP
pJwmgV89TYjcTRROinDCUijL3O+eY5U1957/RoOIq5mRPALX3Kin8fYD/w9R85hVuGpzKzOiC8oV
dPSyI6qB4jHE1b13nz11zltZaFSX1tLIcaRTqymf46deB05hZeUhBjGBqRcFZ2XkTb8hmPQXCMP0
tfrRiWC4H2BiBEq5zGkSOGJoLiorU6aBWik6E8Rdxl/gYWG3ds790ppnbBuHeVHwr+HA0EfQlKAc
q/E8AntpuJ1obUmOUEnfKYWm3NSt+oNhgoiNKKE+jOcy30xMx7eHGTLe33vIVOBzpxG8omhwPoNo
PUmADruvHqaAVsAj5wgV2JksOsiBgWJZJ5ceoZqpB76oTzAbgKgLn8+qwLOWhwi4O5KK3uWHx6y8
xGhbXvEVnkl2qk3YyJxYRuKNpjChsznCTyZ4PfeDxdm8DFXuCDIZbZWg3Yr6/kuYYAypBkmDrMiE
4iIZlSOUreq0/wTGBllkoFAYcVIzkYjT/nQq/MW2bYwSTILXX3uG2etAsP61m91rJ9MZK6x94QmG
YZNAZtZGfFMR1gOijgJD1tiHFaPr3/XjtiadSQ2LOpiSg+seT5OMoANJeQvIZNTBh4134wCOLGE6
SCzdOZStqOnvfsWpi8GtD3xaP8XckU4UJCYzye+A
